bridge/bond: New property `Interface.COPY_MAC_FROM`

cathay4t opened the pull request:
Introducing `Interface.COPY_MAC_FROM` allowing bridge or bond
to use MAC address from specified port.

Example:

```yaml
---
interfaces:
- name: bond99
  type: bond
  state: up
  copy-mac-from: eth1
  link-aggregation:
    mode: balance-rr
    port:
    - eth2
    - eth1
```

Integration test case included.

ffmancera reviewed (changes requested):
I do not see anything wrong on the code but CI linux bridge test is failing.

cathay4t committed:
bridge/bond: New property `Interface.COPY_MAC_FROM`

Introducing `Interface.COPY_MAC_FROM` allowing bridge or bond
to use MAC address from specified port.

Example:

```yaml
---
interfaces:
- name: bond99
  type: bond
  state: up
  copy-mac-from: eth1
  link-aggregation:
    mode: balance-rr
    port:
    - eth2
    - eth1
```

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
> I do not see anything wrong on the code but CI linux bridge test is failing.

The nm plugin is using `original_desire` to detemine whether to create wired setting. The old patch does not include mac address in original desire, so nm does not create one which lead to this failure.

Fixed.

ffmancera commented:
> > I do not see anything wrong on the code but CI linux bridge test is failing.
> 
> The nm plugin is using `original_desire` to detemine whether to create wired setting. The old patch does not include mac address in original desire, so nm does not create one which lead to this failure.
> 
> Fixed.

Thanks for investigating.

ffmancera reviewed (approved):
LGTM, thanks!

smyle10 commented:
If we don't set "copy-mac-from property", what will happen?
Shall we consider implement this property to team interface?

cathay4t commented:
> If we don't set "copy-mac-from property", what will happen?

Systemd-udev will determine the mac address for you. Which means for RHEL 8 and Fedora 32-, it will use mac of first attached port. For Fedora 32+, systemd-udev will generate a random mac regardless the port attached or not.

> Shall we consider implement this property to team interface?

We don't have real user using team interface. So I only touch it when someone need it.

mel1nn commented:
Hi, why this feature is not available for ovs interface? Thanks
